pex/dist_metadata.py

```diff
@@ -7,18 +7,21 @@
 import email
 
 from pex.third_party.packaging.specifiers import SpecifierSet
-from pex.third_party.pkg_resources import DistInfoDistribution
+from pex.third_party.pkg_resources import DistInfoDistribution, Distribution
+from pex.typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from typing import Optional
 
 
 def requires_python(dist):
+    # type: (Distribution) -> Optional[SpecifierSet]
```

Comment:
Note that MyPy doesn't actually know what these are due to the vendoring. But I believe that it will at least check we are using type A, rather than type Z; it just doesn't understand the methods/fields on the type.

And this helpful for the reader.

Reply:
We can possibly create stubs for these if needed?

Reply:
Oh, that'd be interesting as a followup once we have higher coverage. `pkg_resources` is pretty key to Pex. Having type hints working with Pex would be great.